comp/trace/config/config_mock.go

```diff
@@ -0,0 +1,36 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build test
+// +build test
+
+package config
+
+import (
+	"testing"
+
+	pkgconfig "github.com/DataDog/datadog-agent/pkg/config"
+)
+
+// newMock exported mock builder to allow modifying mocks that might be
+// supplied in tests and used for dep injection.
+func newMock(deps dependencies, t testing.TB) (Component, error) {
+	// injected Agent config should be a mock
+
+	traceCfg, err := setupConfig(deps, "apikey")
```

Comment:
Will the config be cleaned after the test is ran ?

Reply:
I'll check this.